internal/plugin/process_monitor.go

```diff
@@ -11,31 +11,25 @@ import (
 	"time"
 
 	"github.com/nginx/agent/v3/internal/bus"
+	"github.com/nginx/agent/v3/internal/config"
 	"github.com/nginx/agent/v3/internal/datasource/host"
 	"github.com/nginx/agent/v3/internal/model"
 )
 
 type GetProcessesFunc func() ([]*model.Process, error)
 
-type ProcessMonitorParameters struct {
-	MonitoringFrequency time.Duration
-	getProcessesFunc    GetProcessesFunc
-}
-
 type ProcessMonitor struct {
-	params      *ProcessMonitorParameters
-	processes   []*model.Process
-	messagePipe bus.MessagePipeInterface
+	monitoringFrequency time.Duration
+	processes           []*model.Process
+	messagePipe         bus.MessagePipeInterface
+	getProcessesFunc    GetProcessesFunc
```

Comment:
I forget, what is the purpose of the `GetProcessesFunc`, why don't we just call `host.GetProcesses()` directly? Is the idea to pass a different `GetProcessesFunc` for K8s?

Reply:
The only purpose of the `GetProcessesFunc` is so that I can mock the call in the unit tests